Windows/Debugger/Debugger_Disasm.cpp

```diff
@@ -912,3 +993,46 @@ void CDisasm::UpdateDialog(bool _bComplete)
 	InvalidateRect (GetDlgItem(m_hDlg, IDC_DEBUGMEMVIEW), NULL, TRUE);
 	UpdateWindow (GetDlgItem(m_hDlg, IDC_DEBUGMEMVIEW));
 }
+
+// More trace logger changes
+void CDisasm::writeToTraceLogger(int writeCode) {
+	CtrlDisAsmView* ptr = CtrlDisAsmView::getFrom(GetDlgItem(m_hDlg, IDC_DISASMVIEW));
+	u32 currentPc = cpu->GetPC();
+	u32 newAddress = currentPc + ptr->getInstructionSizeAt(currentPc);
+	traceLogger.open(traceLogPath.c_str(), std::ios::app);
```

Comment:
I don't think you should keep opening and closing the file at every write, we have our internal class for handling path and file btw, but I guess as this is windows only might be fine using std one directly.

Reply:
If you don't close it after every write then it won't update until the file until you close the disassembly window. Which seems like better functionality in my mind?

Reply:
You may call `traceLogger.flush();` to get the changes updated on disk without closing the file.

Reply:
What is the functional difference between opening/closing constantly and using the flush command and closing at the end, out of curiosity? Like what makes one more reliable or something than the other?

Reply:
Well I expect open/close to be slower (as it have to do more operation), but mostly is about how easy is the code to read, I mean if you want to save the changes to disk that operation is called flush and is better to use it rather than making other to achieve the same result.

Just to note in general open/close will reset the tell position while flush won't (in this case you are always appending so doesn't matter here).

Reply:
okay, i will fix this too. Thank you for the explanation!